tsconfig.json

```diff
@@ -0,0 +1,12 @@
+{
+  "compilerOptions": {
+    "target": "ESNext",
+    "module": "ESNext",
+    "moduleResolution": "node",
+    "strict": true,
+    "declaration": true,
+    "noImplicitOverride": true,
+    "noUnusedLocals": true,
```

Comment:
I'm not sure this needs to be enforced on TS level, ESLint will catch it anyway, and failing compilation due to this can be annoying while developing/debugging